### PR TITLE
ADD: search UI

### PR DESCRIPTION
**Pull request by co012**
Screenshot 2022-08-24 at 17 43 09
Screenshot 2022-08-24 at 17 46 21

### CODE REVIEW

File: packages/app/ui/recently-added-list.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+type BeatEntry = {
+  imageUrl: string
+  videoUrl: string
+  description: string
+  title: string
+  id: string
+  artist: string
+  code: string
+  issuer: string
+}
```
Comment (jmysliv) [inline packages/app/ui/recently-added-list.tsx:20]: move it to models

File: packages/app/ui/recently-added-list.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+export default function RecentlyAddedList() {
+  const [nextPage, setNextPage] = useState(0)
+  const [data, setData] = useState<BeatEntry[]>([])
```
Comment (jmysliv) [inline packages/app/ui/recently-added-list.tsx:24]: I guess we would have a similar use case with pagination in other places as well. We could create a custom hook for that. I can help you with it, or I can add it afterward.

File: packages/app/ui/recently-added-list.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      )}
+    </Pressable>
+  )
+}
+
+function Price({
+  className,
+}: {
+  code?: string
+  issuer?: string
```
Comment (jmysliv) [inline packages/app/ui/recently-added-list.tsx:100]: I think we can remove it if it's unused.

File: packages/app/ui/recently-added-list.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  className?: string
+}) {
+  const [value] = useState(1)
```
Comment (jmysliv) [inline packages/app/ui/recently-added-list.tsx:103]: do we need this `useState`?

File: packages/app/design-system/safe-area-view/index.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { View } from 'app/design-system'
+import { useSafeArea } from 'app/provider/safe-area/use-safe-area'
+import React, { ForwardedRef } from 'react'
+import { ViewProps } from 'app/design-system/view'
+import { View as rView } from 'react-native'
+
+export const SafeAreaView = React.forwardRef(function SafeAreaView(
```
[the remark this answers is not in the document]
Comment (co012) [reply packages/app/design-system/safe-area-view/index.tsx:7]: image
But safeAreaView from react-native-safe-area-context looks alright

File: packages/app/design-system/safe-area-view/index.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  { className, ...rest }: ViewProps,
+  ref: ForwardedRef<rView>
+) {
```
[the remark this answers is not in the document]
Comment (co012) [reply packages/app/design-system/safe-area-view/index.tsx:10]: suggested code is giving me ESLint: Component definition is missing display name(react/display-name), so for now i will leave it like this.

File: packages/app/provider/navigation/index.web.tsx [hunk 7]
```diff
@@ -2,8 +2,24 @@
...
+import Navbar from 'app/ui/navbar'
+import DashboardTabBar from 'app/ui/dashboard-tab-bar'
+
 export const NavigationProvider = ({
   children,
 }: {
   children: React.ReactElement
-}) => <>{children}</>
+}) => {
+  return (
```
[the remark this answers is not in the document]
Comment (co012) [reply packages/app/provider/navigation/index.web.tsx]: Added a condition so naw it displays only on dashboard, but i agree, it is not a great place for it, I placed it here so it won't loose state on dashboard page change.

[the remark this answers is not in the document]
Comment (co012) [reply packages/app/provider/navigation/index.web.tsx]: Added a condition so now it displays only on dashboard, but i agree, it is not a great place for it, I placed it here so it won't loose state on dashboard page change.

[the remark this answers is not in the document]
Comment (jmysliv) [reply packages/app/provider/navigation/index.web.tsx]: if you don't want to lose state, you can use `recoil` and add an atom. However, in this case, the state probably comes straight from the navigation state.

**Pull request closed without merging**